hail/src/main/scala/is/hail/types/physical/PCanonicalArray.scala

```diff
@@ -444,6 +444,12 @@ final case class PCanonicalArray(elementType: PType, required: Boolean = false)
   private def deepRenameArray(t: TArray): PArray =
     PCanonicalArray(this.elementType.deepRename(t.elementType), this.required)
 
+  def constructEmpty(cb: EmitCodeBuilder, region: Value[Region]): SIndexablePointerCode = {
```

Comment:
I don't think there needs to be a separate function for this (and I want to remove the usage in groupByKey above)